docs/src/examples/Accordion.tsx

```diff
@@ -0,0 +1,151 @@
+import {
+  Image,
+  StyleSheet,
+  Text,
+  TouchableOpacity,
+  View,
+  ViewProps,
+} from 'react-native';
+import Animated, {
+  SharedValue,
+  useAnimatedStyle,
+  useDerivedValue,
+  useSharedValue,
+  withTiming,
+} from 'react-native-reanimated';
+import React from 'react';
+
+type AccordionProps = ViewProps & {
+  isExpanded: SharedValue<boolean>;
+  viewKey: number | string;
+  duration?: number;
+};
+
+const AccordionItem = ({
+  isExpanded,
+  children,
+  viewKey,
+  style,
+  duration = 500,
+}: AccordionProps) => {
+  const height = useSharedValue(0);
+
+  const derivedHeight = useDerivedValue(() =>
+    withTiming(height.value * Number(isExpanded.value), {
+      duration,
+    })
+  );
+  const bodyStyle = useAnimatedStyle(() => ({
+    height: derivedHeight.value,
+  }));
+
+  return (
+    <Animated.View
+      key={`accordionItem_${viewKey}`}
+      style={[
+        {
+          width: '100%',
+          overflow: 'hidden',
+        },
+        bodyStyle,
+        style,
+      ]}>
+      <View
+        onLayout={(e) => {
+          height.value = e.nativeEvent.layout.height;
+        }}
+        style={{ width: '100%', position: 'absolute' }}>
+        {children}
+      </View>
+    </Animated.View>
+  );
+};
+
+const Item = () => {
+  return (
+    <Image
+      source={{
+        uri: 'https://upload.wikimedia.org/wikipedia/commons/8/8c/Manul_close.jpg',
+      }}
+      style={styles.image}
+    />
+  );
+};
+
+type ParentProps = {
+  open: SharedValue<boolean>;
+};
+
+const Parent = ({ open }: ParentProps) => {
+  return (
+    <View style={styles.parent}>
+      <AccordionItem isExpanded={open} viewKey="Accordion">
+        <Item />
+      </AccordionItem>
+    </View>
+  );
+};
+
+const AccordionScreen = () => {
+  const open = useSharedValue(false);
+  const onPress = () => {
+    open.value = !open.value;
+  };
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.topPart}>
+        <TouchableOpacity style={styles.button} onPress={onPress}>
+          <Text style={styles.buttonText}>Click me</Text>
+        </TouchableOpacity>
```

Comment:
tbh you can just use the built-in `<Button/>` component from React Native, that would shorten the examples a bit